validator/execution_state.go

```diff
@@ -12,10 +12,11 @@ import (
 )
 
 type GoGlobalState struct {
-	BlockHash  common.Hash
-	SendRoot   common.Hash
-	Batch      uint64
-	PosInBatch uint64
+	BlockHash     common.Hash
+	SendRoot      common.Hash
+	Batch         uint64
+	PosInBatch    uint64
+	HotShotHeight uint64
```

Comment:
The block height of hotshot is used to enforce a continuous mapping between Espresso and nitro blocks (barring the Escape hatch being turned on).